glustercli/cmd/root.go

```diff
@@ -21,36 +34,51 @@ var RootCmd = &cobra.Command{
 			fmt.Println("Error initializing log file ", err)
 		}
 
-		if flagAuthFile == "" && flagSecret == "" {
-			data, err := ioutil.ReadFile(defaultAuthPath)
-			if err != nil && !os.IsNotExist(err) {
-				if verbose {
-					log.WithError(err).Error("failed to read secret")
-				}
-			}
-			secret = string(data)
+		// Secret is taken in following order of precedence (highest to lowest):
+		// --secret
+		// --secret-file
+		// GLUSTERD2_AUTH_SECRET (environment variable)
+		// --secret-file (default path)
+
+		// NOTE: For simplicity, we don't distinguish between an empty
+		// value and an unset value.
+
+		// --secret
+		if flagSecret != "" {
+			secret = flagSecret
 		}
 
-		if flagAuthFile != "" {
-			data, err := ioutil.ReadFile(flagAuthFile)
+		// --secret-file
+		if flagSecretFile != "" && secret == "" {
+			data, err := ioutil.ReadFile(flagSecretFile)
 			if err != nil {
-				if verbose {
-					log.WithError(err).Error("failed to read secret")
-				}
-				failure("failed to read secret", err, 1)
+				failure(fmt.Sprintf("failed to read secret file %s", flagSecretFile),
+					err, 1)
 			}
 			secret = string(data)
 		}
 
-		if flagSecret != "" {
-			secret = flagSecret
+		// GLUSTERD2_AUTH_SECRET
+		if secret == "" {
+			secret = os.Getenv("GLUSTERD2_AUTH_SECRET")
+		}
+
+		// --secret-file (default path)
+		if flagSecretFile == "" && secret == "" {
+			data, err := ioutil.ReadFile(defaultAuthPath)
+			if err != nil && !os.IsNotExist(err) {
+				failure(fmt.Sprintf("failed to read default secret file %s", defaultAuthPath),
```

Comment:
Is there a warning-type utility function? It would probably be worth reporting a problem like a permissions error but only failing if both secret is missing and restauth is true.

Reply:
> Set secret = ""

I can do that :+1:  I don't prefer to ignore permission error altogether. I'll print a warning/info here and not fail. I can shield this printing behind the verbose flag.

Reply:
> I'll print a warning/info here and not fail. I can shield this printing behind the verbose flag

Check old file(left side Line 27-29), which is doing the same.